### ISSUE TITLE
Strange behaviour of `gd` in submodule

### ISSUE THREAD

**Issue by rightx2** [scmbreeze/scm_breeze#309]
- Ubuntu 20.04
- git version: git version 2.30.0
- scm_breeze: master(0a687dc )

I have the main project repository which has a submodule. Below is the directory structure in my workspace
- main: `~/Dropbox/Programming/Workspace/django/super_trading_project`
- submodule: `~/Dropbox/Programming/Workspace/django/super_trading_project/super_trading/backtester/event_driven`
- `cat` of `....../event_driven/.git`: `gitdir: ../../../.git/modules/super_trading/backtester/event_driven`

## Problem
When the change in the file of `event_driven` occurs and type `gs` it looks like this:
```
➤ Changes not staged for commit
#
#      modified: [1] ../../../.git/modules/super_trading/backtester/event_driven/alphas/mpaa.py
```

When I run just `gd`, it shows all diff information and it works well

But the problem is happening when I run `gd [number]`. `gd 1` shows an error like this:

```
=> gd 1
fatal: ambiguous argument '~/Dropbox/Programming/Workspace/django/super_trading_project/.git/modules/super_trading/backtester/event_driven/alphas/mpaa.py': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

I have no idea why it happens!

Some notes:

- `gd [number]` works well in parent modules. It does not work only in git submodule
- All these strange things don't happen in Mac OS X
- I've faced with the same problem even when I clone the new parent module and submodule in the clean directory

**Comment by rightx2** [comment]
If I run with a raw command, `git diff file_name`(without `scm_breeze`), it works well..

**Comment by lrntgr** [comment]
I'm facing the same issue here. From within the submodule directory:
 - regular `git status` works correctly
    ```
    $ git status
    Untracked files:
      (use "git add <file>..." to include in what will be committed)
	    foo.txt
    ```
 - SCM breeze `gs` displays a weird `/.git/modules` path
    ```
    $ gs
    # On branch:   |  [*] => $e*
    #
    ➤ Untracked files
    #
    #     untracked: [1] ../.git/modules/<SUBMODULE_DIRNAME>/foo.txt
    #
    ```
As a result, `gd 1` fails due to this abnormal path : `fatal: ambiguous argument '<REPO_DIRNAME>/.git/modules/<SUBMODULE_DIRNAME>/foo.txt': unknown revision or path not in the working tree.`, whereas `git diff foo.txt` works well.

My environment is:
```
$ lsb_release -a 
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.1 LTS (fossa-cory-tgl X59)
Release:	20.04
Codename:	focal

$ git --version
git version 2.25.1

$ cd $HOME/.scm_breeze && git rev-parse --short HEAD
ff6af42
```

**Comment by lrntgr** [comment]
I solved the issue by installing `ruby`.

Indeed, 'scm_breeze.sh' behaves differently depending on whethter `ruby` is present or not (cf [code permalink](https://github.com/scmbreeze/scm_breeze/blob/ff6af425e2769c55061d866c1e8fb4def4efab12/scm_breeze.sh#L30:L34)):
> ```shell
>  if ! type ruby > /dev/null 2>&1; then
>    # If Ruby is not installed, fall back to the
>    # slower bash/zsh implementation of 'git_status_shortcuts'
>    source "$scmbDir/lib/git/fallback/status_shortcuts_shell.sh"
>  fi
>```

- with `ruby` installed, `gs` is aliased to the `git_status_shortcuts` defined in **status_shortcuts.sh**, which works properly.
- without `ruby` installed, `gs` is aliased to the `git_status_shortcuts` defined in **status_shortcuts_shell.sh**, which seems to be buggy (?).

**Comment by ghthor** [comment]
Yes, the shell version needs to be worked on and is known to have some
strange behavior. Glad you were able to you issue by installing ruby.
